### ISSUE TITLE
"Failed to locate a package.json file that has a "tree-sitter" section, please ensure you have one, and if you don't then consult the docs"

### ISSUE THREAD

**Issue by olehermanse** [tree-sitter/tree-sitter#3283]
### Problem

I get this error message:

```
Failed to locate a package.json file that has a "tree-sitter" section, please ensure you have one, and if you don't then consult the docs
```

By following the tutorial in the docs:

https://tree-sitter.github.io/tree-sitter/creating-parsers

I tried googling a bit, searching previous issues in GitHub, and scrolling through the docs site, but I did not find any good explanation of why this happens or what to do.

I ended up finding another tree-sitter repo and copying / translating what they had in their package.json, I put this in my `package.json`:

```
  "tree-sitter": [
    {
      "scope": "source.cfengine",
      "file-types": [
        "cf"
      ]
    }
  ],
```

PS: I think I solved it, I don't need help, consider this more of a suggestion for how to improve the docs / tooling.

### Steps to reproduce

Follow the tutorial for creating parsers at: https://tree-sitter.github.io/tree-sitter/creating-parsers

### Expected behavior

No error messages while following the steps in the tutorial. Or clear instructions about what to do in this case.

### Tree-sitter version (tree-sitter --version)

tree-sitter 0.22.2 (fc15f621334a262039ffaded5937e2844f88da61)

### Operating system/version

macOS 14.4.1

**Comment by WillLillis** [comment]
I believe this has been addressed by https://github.com/tree-sitter/tree-sitter/pull/3224

**Comment by olehermanse** [comment]
Merged yesterday, nice :) Waiting for 0.22.3 then

**Comment by verhovsky** [comment]
I get this error when running

```sh
npx tree-sitter build --wasm node_modules/tree-sitter-bash --output dist/tree-sitter-bash.wasm
```

after upgrading

```diff
-    "tree-sitter-bash": "^0.21.0",
-    "web-tree-sitter": "^0.22.6",
+    "tree-sitter-bash": "^0.23.1",
+    "web-tree-sitter": "^0.24.3",
```

I'm just using the parser from tree-sitter-bash through WASM to serve in the browser 

https://github.com/curlconverter/curlconverter/blob/767bda3a22511a09191b85b026fa53ab066d8bee/src/shell/webParser.ts#L13

why do I need to add stuff to my package.json now?

When I run 

```sh
npx tree-sitter init
```

It asks me for the name of the parser, but I'm not making a parser.
